File: backend-service/src/main/java/com/getyourguide/backendservice/bean/Activity.java
package com.getyourguide.backendservice.bean;


import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Data
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Activity {
    private Long id;
    private String title;
    private int price;
    private String currency;
    private double rating;
    private boolean specialOffer;
    private int supplierId;
}

File: frontend-service/src/app/activity/activity.ts
export class Activity {
    constructor(
        public id: number,
        public title: string,
        public price: number,
        public currency: string,
        public rating: number,
        public specialOffer: boolean,
        public supplierId: number
        ) {

        }
    
}


export class ActivityModel {
    private activities: Activity[];
    constructor() {
        this.activities = new Array<Activity>();
    }

    addActivity(activity: Activity) {
        this.activities.push(activity);
    }

    filterActivityByTitle(title: string){
        title = title.toUpperCase();
        return this.activities.filter(activity => {
            let titleToFilter = activity.title.toUpperCase();
            if (titleToFilter == title ||
                titleToFilter.startsWith(title) ||
                titleToFilter.endsWith(title) ||
                titleToFilter.includes(title))
                return true;
            else
                return false;
                }
        );
    }
}